src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -91,10 +101,47 @@ def load(self) -> list[Rule]:
             msg = "Please run: databricks labs ucx table-mapping"
             raise ValueError(msg) from None
 
-    def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str):
+    def get_tables_to_migrate(self, tables_crawler: TablesCrawler):
+        upgraded_tables = self._get_upgraded_tables()
+
+        tables_to_validate = []
+        rules = self.load()
+        crawled_tables = tables_crawler.snapshot()
+        crawled_tables_keys = {crawled_table.key: crawled_table for crawled_table in crawled_tables}
+        # Deducing the set of databases in the crawled tables
+        crawled_databases = {crawled_table.database for crawled_table in crawled_tables}
+
+        # Getting all the source tables from the rules
+        source_databases = {rule.src_schema for rule in rules if rule.src_schema in crawled_databases}
+        validated_databases = self._get_validated_databases(source_databases)
+
+        for rule in rules:
+            if rule.as_hms_table_key not in crawled_tables_keys:
+                logger.info(f"Table {rule.as_hms_table_key} in the mapping doesn't show up in assessment")
+                continue
+            if rule.as_uc_table_key in upgraded_tables:
+                logger.info(f"Table {rule.as_hms_table_key} was migrated to {rule.as_uc_table_key} and will be skipped")
+                continue
+            if rule.src_schema not in validated_databases:
+                logger.info(f"Table {rule.as_hms_table_key} is in a database that was marked to be skipped")
+                continue
+            tables_to_validate.append(crawled_tables_keys[rule.as_hms_table_key])
+        validated_tables = self._get_validated_tables(tables_to_validate)
+        validated_tables_by_key = {table.key: table for table in validated_tables}
+
+        tables_to_migrate = []
```

Comment:
Lines 118-137 are not necessary if you construct TableToMigrate from a thread, which gets both Table and Rule as arguments. Do that also for tables.

Reply:
Will Do